### PR TITLE
added linkcode_resolve

### PR DESCRIPTION
**Pull request by selvaraj-sembulingam**
Changed _sphinx.ext.viewcode_ to _sphinx.ext.linkcode_

### CODE REVIEW

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @selvaraj-sembulingam, great work 🚀🚀🚀

I have some comments on how the code is structured that I would like you to fix before we merge. Also the support of different module names.

File: docs/conf.py [hunk 1]
```diff
@@ -150,3 +151,58 @@
...
+def linkcode_resolve(domain, info):
+    if domain != "py":
+        return None
+    if not info["module"]:
+        return None
+
+    modname = info["module"]
+    fullname = info["fullname"]
+
+    submod = sys.modules.get(modname)
```
[the remark this answers is not in the document]
Comment (selvaraj-sembulingam) [reply docs/conf.py]: Done

File: docs/conf.py [hunk 2]
```diff
@@ -150,3 +151,58 @@
...
+    if submod is None:
+        return None
+
+    obj = submod
```
[the remark this answers is not in the document]
Comment (selvaraj-sembulingam) [reply docs/conf.py]: Removed

File: docs/conf.py [hunk 3]
```diff
@@ -150,3 +151,58 @@
...
+        unwrap = inspect.unwrap
+    except AttributeError:
+        pass
+    else:
+        obj = unwrap(obj)
+
+    fn = None
+    lineno = None
+
+    if fn is None:
```
[the remark this answers is not in the document]
Comment (selvaraj-sembulingam) [reply docs/conf.py]: Removed

File: docs/conf.py [hunk 4]
```diff
@@ -150,3 +151,58 @@
...
+        source, lineno = inspect.getsourcelines(obj)
+    except Exception:
+        lineno = None
+
+    fn = os.path.relpath(fn, start=os.path.dirname(__import__("ivy").__file__))
+
+    if lineno:
+        linespec = "#L%d-L%d" % (lineno, lineno + len(source) - 1)
+    else:
+        linespec = ""
```
[the remark this answers is not in the document]
Comment (selvaraj-sembulingam) [reply docs/conf.py]: Done

File: docs/conf.py [hunk 5]
```diff
@@ -150,3 +151,58 @@
...
+
+    github_url = f"https://github.com/unifyai/ivy/blob/master/ivy/{fn}{linespec}"
+
+    return github_url
```
[the remark this answers is not in the document]
Comment (selvaraj-sembulingam) [reply docs/conf.py]: Done

File: docs/conf.py [hunk 6]
```diff
@@ -150,3 +151,58 @@
...
+    fn = None
+    lineno = None
+
+    if fn is None:
+        try:
+            fn = inspect.getsourcefile(obj)
+        except Exception:
+            fn = None
+        if not fn:
+            return None
```
[the remark this answers is not in the document]
Comment (selvaraj-sembulingam) [reply docs/conf.py]: Yeah Done

File: docs/conf.py [hunk 7]
```diff
@@ -150,3 +151,51 @@
...
+
+    try:
+        source, lineno = inspect.getsourcelines(modname)
+    except Exception:
+        lineno = None
+
+    fn = os.path.relpath(fn, start=os.path.dirname(__import__(mod_name).__file__))
+
+    if lineno:
+        linespec = "#L%d-L%d" % (lineno, lineno + len(source) - 1)
```
Comment (KareemMAX) [inline docs/conf.py:198]: Don't you think that might return an exception in case of `lineno` is `None`?

Comment (KareemMAX) [inline docs/conf.py:198]:
```suggestion
    linespec = ""

    if lineno:
        linespec = "#L%d-L%d" % (lineno, lineno + len(source) - 1)
```

File: docs/conf.py [hunk 8]
```diff
@@ -150,3 +151,58 @@
...
+            fn = None
+        if not fn:
+            return None
+
+    try:
+        source, lineno = inspect.getsourcelines(obj)
+    except Exception:
+        lineno = None
+
+    fn = os.path.relpath(fn, start=os.path.dirname(__import__("ivy").__file__))
```
[the remark this answers is not in the document]
Comment (KareemMAX) [reply docs/conf.py]: I'm not really sure if we should be that coupled to the naming convention. Wouldn't it be better if we got `repo_name` from the partial config?

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Awesome work, please address my changes and then we are good to go

File: docs/conf.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (selvaraj-sembulingam) [reply docs/conf.py]: Sure

**Pull request merged**